tests/robot-tests/reports.py

```diff
@@ -79,36 +105,48 @@ def _merge_test_reports(test_results_folder):
 
 
 def filter_out_passing_suites_from_report_file(path_to_original_report: str, path_to_filtered_report: str):
-    
-    with open(path_to_original_report, "rb") as report:
-        
-        report_contents = report.read()
+    with open(path_to_original_report, "rb") as report_file:
+        report_contents = report_file.read()
         report = BeautifulSoup(report_contents, features="xml")
-        
+
         passing_suite_ids = _get_passing_suite_ids_from_report(report)
-        
-        suite_elements = report.find_all('suite', recursive=True)
-        [suite_element.extract() for suite_element in suite_elements if suite_element.get('id') in passing_suite_ids]
-        
-        suite_stats = report.find_all('stat', recursive=True)
-        [suite_stat.extract() for suite_stat in suite_stats if suite_stat.get('id') in passing_suite_ids]
-    
+
+        suite_elements = report.find_all("suite", recursive=True)
+        [suite_element.extract() for suite_element in suite_elements if suite_element.get("id") in passing_suite_ids]
+
+        suite_stats = report.find_all("stat", recursive=True)
+        [suite_stat.extract() for suite_stat in suite_stats if suite_stat.get("id") in passing_suite_ids]
+
         if os.path.exists(path_to_filtered_report):
             os.remove(path_to_filtered_report)
 
         with open(path_to_filtered_report, "a") as filtered_file:
             filtered_file.write(report.prettify())
 
-        
-def _get_passing_suite_ids_from_report(report: BeautifulSoup) -> []: 
-    suite_results = report.find('statistics').find('suite').find_all('stat')
-    passing_suite_results = [suite_result for suite_result in suite_results if int(suite_result.get('fail')) == 0]
-    return [passing_suite.get('id') for passing_suite in passing_suite_results]
+
+def _get_passing_suite_ids_from_report(report: BeautifulSoup) -> []:
+    suite_results = report.find("statistics").find("suite").find_all("stat")
+    passing_suite_results = [suite_result for suite_result in suite_results if int(suite_result.get("fail")) == 0]
+    return [passing_suite.get("id") for passing_suite in passing_suite_results]
+
+
+def _get_failing_leaf_suite_ids_from_report(report: BeautifulSoup) -> []:
```

Comment:
I'm not sure what you mean by `leaf` here

Reply:
See above :)